### PR TITLE
fix: fixing excluded directories option

### PR DESCRIPTION
**Pull request by area73**
When using option object with isExcludedDir, promises are never resolved.
* Also adding a missing test for this use case.
* Adding a prettier config with the project's coding style

### CODE REVIEW

**Comment by area73** [comment]
First I try to do a  `notifier.done = true` on line 117 and that resolves de promise but then I realized that we need to subtract one pending element.

So I did it with `--pending`. although functions is working as expected I can't pass the test now because it says "Promise returned by test never resolved" , but I couldn't find how to fix the test

**Comment by TomerAberbach** [comment]
Thanks for looking into this!

Shot in the dark: maybe the issue occurs if the last directory we search is excluded. In this case it looks like the code might never resolve the notifier promise? So it gets stuck on `await notifier.onResolved()`?

Not immediately sure of the fix, but I can think about it more in the new year

**Comment by area73** [comment]
So, you were right! 🥳 🥳.
I didn't think about that particular use case 😬
Thanks!

**Comment by TomerAberbach** [comment]
Nice! Although, I think there might still be a bug if there's more than one directory and they're all excluded (I haven't tested, but it looks that way because `pending` would be greater than one).

Looking at the code again, I think we can improve the structure a bit and fix the bug.

The `pending` variable should represent the number of pending promises. How about instead of setting `pending` to `dirnames.length` initially, we do this instead?
```js
let pending = 0

for (const dirname of dirnames) {
  if (options.isExcludedDir(dirname)) {
    continue
  }
 
  pending++
  // Rest of the code in the loop...
}

if (pending === 0) {
  notifier.done = true
}
```

This way `pending` ends up being the exact number of promises we schedule. If `pending` is zero just after the loop, then that means we did not schedule any promises and we can mark the notifier as done. Does that work?

File: test/index.js [hunk 1]
```diff
@@ -20,6 +20,13 @@ import { getAllFiles, getAllFilesSync } from '../src/index.js'
 
 const fixtures = `./test/fixtures`
 
+const isInList = list => name => list.indexOf(name) > -1
```
Comment (TomerAberbach) [inline test/index.js]: I think this can be more simply written as `list => name => list.includes(name)`

**Comment by area73** [comment]
👍

**Comment by TomerAberbach** [comment]
Thanks for making the changes! Seems the tests are failing. Seems related to switching to 8 files in the tests? I only counted 6 in your PR. Did you forget to commit some files you added locally?

**Comment by area73** [comment]
Bummer forgot to add the new files  😬

**Comment by TomerAberbach** [comment]
Seems the tests are failing on Windows. I wonder if it's because the Windows paths are returned with backslashes, but the excluded dir array has it using forward slashes. Might need to use `path.posix.normalize` before the includes (I _think_ that's what method is called)

**Comment by ajvincent** [comment]
Possibly related crash in NodeJS using isExcludedDir: https://github.com/TomerAberbach/get-all-files/issues/8

**Comment by area73** [comment]
So I had a  hard time trying to fix this issue, I did a lot of testing on CI and finally, I think I came up with some sort of solution, maybe I'm still missing some use case but now test passes.
It has been a nightmare trying to pass correct  posix (slash, backslash, and also './' was an issue)

**Comment by TomerAberbach** [comment]
Thanks for all this work! I don't have bandwidth to look at the new changes today, but I'll try to take a look as soon as I can find the time

**Comment by TomerAberbach** [comment]
I rewrote the async getAllFiles function so this PR is obsolete now

**Pull request closed without merging**